### PR TITLE
Returning an error if a PropType is passed to ReactPropTypes.oneOf.

### PR DESCRIPTION
**Pull request by ventuno**
I took where @philipshurpik left off with https://github.com/facebook/react/pull/6940 to fix https://github.com/facebook/react/issues/1919, as that PR can't be merged anymore and is more than a month old.

@philipshurpik, if you meant to keep on working on https://github.com/facebook/react/pull/6940, please let me know and I'll close this one.

The idea is to return a `PropTypeError` whenever one of `ReactPropTypes` is passed to `ReactPropTypes.oneOf`. The error message, hints that the user might be meaning to use `ReactPropTypes.oneOfType` instead.

1) If `createChainableTypeChecker` is called with an `expectedType` we add the `expectedType` flag to the `checkType` function (see [1])
2) In `createEnumTypeChecker.validate`, if `expectedValues[i]` is of type `function` and it has the `expectedType` flag, we know that a `ReactPropTypes` has been passed, so we throw the error described above
3) As per https://github.com/facebook/react/pull/6940, if a generic `function` is passed to `ReactPropTypes.oneOf` we improve the previous output to display "Unexpected function" and the function converted to string

As a side note, I noticed that if `getPropType` is called with `undefined`, we get the following exception:
`Cannot read property '@@toStringTag' of undefined`
Is this expected?

[1] https://github.com/facebook/react/pull/6940/files#r68511156

### CODE REVIEW

File: src/isomorphic/classic/types/ReactPropTypes.js [hunk 1]
```diff
@@ -295,19 +298,40 @@ function createEnumTypeChecker(expectedValues) {
...
     );
   }
+
+  function functionPrettifier(name, values) {
+    for (var i = 0; i < values.length; i++) {
+      if (getPropType(values[i]) === 'function') {
+        values[i] = 'Unexpected function: ' + values[i].toString();
+      }
+    }
+    return values;
```
Comment (donavon) [inline src/isomorphic/classic/types/ReactPropTypes.js]: maybe something like this instead?
```js
return values.map(value => (
  getPropType(value) === 'function' ? `Unexpected function: ${value}` : value
));
```
You shouldn't have to specially call `toString` on `value`.

Comment (ventuno) [inline src/isomorphic/classic/types/ReactPropTypes.js]: Good idea, @donavon. Thanks a lot, I implemented your recommended change :-).

**Comment by ventuno** [comment]
Dear @aweary and @gaearon, I just noticed there was some activity about this PR [1]. So I decided to fix the merging issues. Can you kindly review?

[1] https://github.com/facebook/react/pull/8274#issuecomment-260143130

**Comment by gaearon** [comment]
Sorry about this! We've been busy with a React rewrite this year, and haven't really paid attention to PRs. :-(

This code has moved to https://github.com/facebook/prop-types so you can make a PR there and we could review.

**Pull request closed without merging**